File: 0383-ransome-note/main.c
#include <string.h>

bool canConstruct(char* ransomNote, char* magazine) {
    int magazineLetters[26] = {0};
    int len = strlen(magazine);

    for (int i = 0; i < len; i++) {
        magazineLetters[magazine[i] - 'a']++;
    }

    len = strlen(ransomNote);
    for (int i = 0; i < len; i++) {
        magazineLetters[ransomNote[i] - 'a']--;

        if (magazineLetters[ransomNote[i] - 'a'] < 0) {
            return false;
        }
    }

    return true;
}

File: 0383-ransome-note/main.go
func canConstruct(ransomNote string, magazine string) bool {
    var magazineLetters = [26]int{}

    for _, ch := range magazine {
        magazineLetters[int(ch) - int('a')]++
    }

    for _, ch := range ransomNote {
        magazineLetters[int(ch) - int('a')]--
        if magazineLetters[int(ch) - int('a')] < 0 {
            return false
        }
    }

    return true
}
